payments/api.py

```diff
@@ -462,3 +475,341 @@ def get_auto_apply_discounts_for_basket(basket_id: int) -> QuerySet[Discount]:
+
+    if (
+        "redemption_type" in kwargs
+        and kwargs["redemption_type"] in ALL_REDEMPTION_TYPES
+    ):
+        redemption_type = kwargs["redemption_type"]
+
+    if "expires" in kwargs and kwargs["expires"] is not None:
+        expiration_date = parse_supplied_date(kwargs["expires"])
+    else:
+        expiration_date = None
+
+    if "activates" in kwargs and kwargs["activates"] is not None:
+        activation_date = parse_supplied_date(kwargs["activates"])
+    else:
+        activation_date = None
+
+    if "integrated_system" in kwargs and kwargs["integrated_system"] is not None:
+        # Try to get the integrated system via ID or slug.  Raise an exception if it doesn't exist.  # noqa: E501
+        # check if integrated_system is an integer or a slug
+        integrated_system_missing_msg = (
+            f"Integrated system {kwargs['integrated_system']} does not exist."
+        )
+        if kwargs["integrated_system"].isdigit():
+            try:
+                integrated_system = IntegratedSystem.objects.get(
+                    pk=kwargs["integrated_system"]
+                )
+            except IntegratedSystem.DoesNotExist:
+                raise ValueError(integrated_system_missing_msg)  # noqa: B904
+        else:
+            try:
+                integrated_system = IntegratedSystem.objects.get(
+                    slug=kwargs["integrated_system"]
+                )
+            except IntegratedSystem.DoesNotExist:
+                raise ValueError(integrated_system_missing_msg)  # noqa: B904
+    else:
+        integrated_system = None
+
+    if "product" in kwargs and kwargs["product"] is not None:
+        # Try to get the product via ID or SKU.  Raise an exception if it doesn't exist.
+        product_missing_msg = f"Product {kwargs['product']} does not exist."
+        if kwargs["product"].isdigit():
+            try:
+                product = Product.objects.get(pk=kwargs["product"])
+            except Product.DoesNotExist:
+                raise ValueError(product_missing_msg)  # noqa: B904
+        else:
+            try:
+                product = Product.objects.get(sku=kwargs["product"])
+            except Product.DoesNotExist:
+                raise ValueError(product_missing_msg)  # noqa: B904
+    else:
+        product = None
+
+    if "users" in kwargs and kwargs["users"] is not None:
+        # Try to get the users via ID or email.  Raise an exception if it doesn't exist.
+        users = []
+        user_missing_msg = "User %s does not exist."
+        for user_identifier in kwargs["users"]:
+            if user_identifier.isdigit():
+                try:
+                    users.append(User.objects.get(pk=user_identifier))
+                except User.DoesNotExist:
+                    raise ValueError(user_missing_msg % user_identifier)  # noqa: B904
+            else:
+                try:
+                    user = User.objects.get(email=user_identifier)
```

Comment:
Tested with an email address that definitely exists in my copy and got an error.